src/Parser/Druid/Feral/Modules/Bleeds/RipSnapshot.js

```diff
@@ -0,0 +1,237 @@
+ */
+
+const RIP_BASE_DURATION = 24000;
+const RIP_POWER_PER_COMBO = 0.20;
+const BITE_EXECUTE_RANGE = 0.25;
+
+class RipSnapshot extends Snapshot {
+  static dependencies = {
+    combatants: Combatants,
+    comboPointTracker: ComboPointTracker,
+  };
+
+  ripCastCount = 0;
+  downgradeCount = 0;
+  shouldBeBiteCount = 0;
+
+  /**
+   * Order of processed events when player casts rip is always:
+   *   on_byPlayer_cast
+   *   on_byPlayer_spendresource
+   *   on_byPlayer_debuffapply, or on_byPlayer_debuffrefresh
+   * So it's safe to store the comboLastRip on spendresource then use it in calcPower,
+   * which is called by Snapshot in response to debuffapply and debuffrefresh.
+   */
+  comboLastRip;
+
+  hasSabertooth = false;
+  healthFraction = {};
+
+  on_initialized() {
+    this.spellCastId = SPELLS.RIP.id;
+    this.debuffId = SPELLS.RIP.id;
+    this.durationOfFresh = RIP_BASE_DURATION;
+    this.isProwlAffected = false;
+    this.isTigersFuryAffected = true;
+    this.isBloodtalonsAffected = true;
+
+    const combatant = this.combatants.selected;
+    if (combatant.hasTalent(SPELLS.JAGGED_WOUNDS_TALENT.id)) {
+      this.durationOfFresh *= JAGGED_WOUNDS_MODIFIER;
+    }
+    if (combatant.hasTalent(SPELLS.SABERTOOTH_TALENT.id)) {
+      this.hasSabertooth = true;
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    super.on_byPlayer_cast(event);
+    if (SPELLS.RIP.id === event.ability.guid) {
+      this.ripCastCount += 1;
+      debug && console.log(`${this.owner.formatTimestamp(event.timestamp)} rip cast.`);
+    }
+    if (SPELLS.FEROCIOUS_BITE.id === event.ability.guid) {
+      debug && console.log(`${this.owner.formatTimestamp(event.timestamp)} bite cast.`);
+      this.handleBiteExtend(event);
+    }
+  }
+
+  on_byPlayer_spendresource(event) {
+    if (SPELLS.RIP.id === event.ability.guid &&
+        event.resourceChangeType === RESOURCE_TYPES.COMBO_POINTS.id) {
+      this.comboLastRip = event.resourceChange;
+    }
+  }
+
+  // use damage events to keep track of each target's health to tell when they're in "execute" range
+  on_byPlayer_damage(event) {
+    // no need to track health if combatant has sabertooth, and don't track health of friendlies.
+    if (this.hasSabertooth || event.targetIsFriendly) {
+      return;
+    }
+    this.healthFraction[encodeTargetString(event.targetID, event.targetInstance)] = event.hitPoints / event.maxHitPoints;
+  }
+
+  biteCanRefresh(target) {
+    return this.hasSabertooth || 
+      (this.healthFraction[target] && this.healthFraction[target] < BITE_EXECUTE_RANGE);
+  }
+
+  handleBiteExtend(event) {
+    const target = encodeTargetString(event.targetID, event.targetInstance);
+    if (!this.biteCanRefresh(target)) {
+      return;
+    }
+    const existing = this.stateByTarget[target];
+    if (!existing || existing.expireTime < event.timestamp) {
+      return;
+    }
+
+    // Bite "refreshes the duration of your Rip on the target" which means setting to the base duration.
```

Comment:
Is this always the case? Say I cast a rip before a pandemic window so I have a 1.3\*duration rip up and it's duration is still > 1\*duration, will it reduce the duration to 1\*duration?